scripts/genproto.sh

```diff
@@ -25,8 +25,25 @@ fi
 source ./scripts/test_lib.sh
 
 if [[ $(protoc --version | cut -f2 -d' ') != "3.20.3" ]]; then
-  echo "could not find protoc 3.20.3, is it installed + in PATH?"
-  exit 255
+  echo "Could not find protoc 3.20.3, installing now..."
+
+  arch=$(go env GOARCH)
+
+  case ${arch} in
+    "amd64") file="x86_64" ;;
+    "arm64") file="aarch_64" ;;
+    *)
+      echo "Unsupported architecture: ${arch}"
+      exit 255
+      ;;
+  esac
+
+  download_url="https://github.com/protocolbuffers/protobuf/releases/download/v3.20.3/protoc-3.20.3-linux-${file}.zip"
+  echo "Running on ${arch}."
+  wget ${download_url} && unzip protoc-3.20.3-linux-${file}.zip
+  mv bin/protoc /usr/local/bin/ && chmod +x /usr/local/bin/protoc
+  echo "export PATH=$PATH:/usr/local/bin" >> ~/.bashrc
```

Comment:
Also think updating `.bashrc` without explicit permission might be a bit invasive.  Let's just run the `export PATH` command directly based on file in working directory rather than adding it to `.bashrc`.

Reply:
+1

Or what about introduce gitignored temporary dir in the repo? Install in the temporary dir and export it in the PATH. No need to worry about .bashrc

Reply:
Updated thanks